### ISSUE TITLE
insertOrUpdate generates invalid code when all columns are part of primary key

### ISSUE THREAD

**Issue by CalumMcCall** [slick/slick#1728]
Hi,
I think this might be a duplicate of #1627. I have a table like this:

``` 
class AutoInstallTable(tag: Tag) extends Table[AutoInstall](tag, "AutoInstall") {
    def namespace = column[Namespace]("namespace")
    def pkgName = column[PackageId.Name]("pkg_name")
    def device = column[Uuid]("device")

    def pk = primaryKey("pk_auto_install", (namespace, pkgName, device))

    def * = (namespace, pkgName, device).shaped <>
      ((AutoInstall.apply _).tupled, AutoInstall.unapply)
}
```
And a mutator method like so:

```
def addDevice(ns: Namespace, pkgName: PackageId.Name, dev: Uuid)
               (implicit ec: ExecutionContext): DBIO[Unit]
    = all.insertOrUpdate(AutoInstall(ns, pkgName, dev)).map(_ => ())
```
When running a query on an empty db with no values, I get the following error:

```
java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '' at line 1
Query is : insert into `AutoInstall` (`namespace`,`pkg_name`,`device`)  values ('default','ss','dec7388e-4466-428f-8f8a-4999f5ddbe45') on duplicate key update 
	at org.mariadb.jdbc.internal.util.ExceptionMapper.get(ExceptionMapper.java:125)
	at org.mariadb.jdbc.internal.util.ExceptionMapper.throwException(ExceptionMapper.java:69)
	at org.mariadb.jdbc.MariaDbStatement.executeQueryEpilog(MariaDbStatement.java:250)
	at org.mariadb.jdbc.MariaDbClientPreparedStatement.executeInternal(MariaDbClientPreparedStatement.java:210)
	at org.mariadb.jdbc.MariaDbClientPreparedStatement.executeUpdate(MariaDbClientPreparedStatement.java:186)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$InsertOrUpdateAction$$anonfun$nativeUpsert$1.apply(JdbcActionComponent.scala:564)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$InsertOrUpdateAction$$anonfun$nativeUpsert$1.apply(JdbcActionComponent.scala:561)
	at slick.jdbc.JdbcBackend$SessionDef$class.withPreparedStatement(JdbcBackend.scala:372)
	at slick.jdbc.JdbcBackend$BaseSession.withPreparedStatement(JdbcBackend.scala:434)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl.preparedInsert(JdbcActionComponent.scala:502)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$InsertOrUpdateAction.nativeUpsert(JdbcActionComponent.scala:561)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$InsertOrUpdateAction.f$1(JdbcActionComponent.scala:544)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$InsertOrUpdateAction.run(JdbcActionComponent.scala:557)
	at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:31)
	at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:28)
	at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:240)
	at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:240)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```
The latter half of the SQL query seems to be missing. This is on mariadb using Slick 3.2.0. This code worked fine using Slick 3.1.1.

**Comment by ievans** [comment]
I experienced a similar error with Slick 3.2.0 against Postgres, where slick generated a query for insertOrUpdate on a table that only contained two columns, both of which were used in the primary key, that caused postgres to complain with: `org.postgresql.util.PSQLException: ERROR: syntax error at or near "where"`. 

Adding a third (non-PK) column fixed the issue.

**Comment by crafty-coder** [comment]
I have the same issues using MariaDB

```
Caused by: org.mariadb.jdbc.internal.util.dao.QueryException: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '' at line 1
Query is: insert into `user_dataset` (`user_id`,`dataset_id`)  values (?,?) on duplicate key update , parameters [2,3]
```

**Comment by mikla** [comment]
+1 Faced the same problem as @ievans

**Comment by hvesalai** [comment]
Anybody able to say if this is aduplicate of #1627 so we could close this?

**Comment by ievans** [comment]
Hey @hvesalai, I saw this GitHub issue was closed, but after upgrading to Slick 3.2.3, I still have this issue. Just wanted to make sure it wasn't closed by mistake.
